packages/easy-coding-standard/src/HttpKernel/EasyCodingStandardKernel.php

```diff
@@ -35,14 +36,10 @@ public function registerBundles(): iterable
 
     public function getCacheDir(): string
     {
-        // the PACKAGE_VERSION constant helps to rebuild cache on new release, but just once
-        $cacheDirectory = sys_get_temp_dir() . '/ecs_' . get_current_user();
+        $cacheDir = sys_get_temp_dir() . '/ecs_' . get_current_user() . '_' . random_int(0, 100000);
+        FileSystem::createDir($cacheDir);
```

Comment:
This will build a new container on each run now, which (performance aside) is a good thing I think. What about instead using the same directory but wipe it before starting? Then you would not end up with new files on each run that cannot be reused anyway. 

For caching: As long as the configuration gets baked into the container, we IMHO either need an environment or a container per configuration. I first thought about a hash that includes the path + filemtime of the config file, but as other files can get used + included this isn't enough. So not trivial… :-/ 

That said: From looking at the code, I think this fixes #3398. I'll probably get to testing it tomorrow. Thanks a lor for the super quick fix! :heart:

Reply:
> What about instead using the same directory but wipe it before starting?

That makes more sense then spamming temp directory with random names... could you send PR?

Reply:
Here you go: https://github.com/symplify/symplify/pull/3403